includes/datastream.inc

```diff
@@ -138,6 +143,7 @@ function islandora_rest_datastream_post_response(array $parameters) {
   $datastream->label = isset($request['label']) ? $request['label'] : $file['name'];
   $datastream->versionable = isset($request['versionable']) ? (bool) $request['versionable'] : TRUE;
   $datastream->mimetype = isset($request['mimeType']) ? $request['mimeType'] : $mime_detect->getMimetype($file['name']);
+  $datastream->checksumType = isset($request['checksumType']) ? $request['checksumType'] : 'DISABLED';
```

Comment:
@qadan Thinking about this a bit, I'm not sure we want to check the list of DSIDs in `variable_get('islandora_checksum_dsids_to_check', '')` after all. The REST API allows requests to override other datastream properties that are constrained in the datastream upload UI, like state, control group, mimetype, and versionable. Seems like we're breaking that pattern here if we only allow generation of checksums on the DSIDs listed in the GUI? We also allow the REST API to override the default checksumType, something you can't do anywhere in the Islandora the GUI.